### PR TITLE
Dashboards: Correctly display Admin access to dashboards in the UI

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

Fixes the permission evaluation in the backend to ensure that dashboard permission UI correctly displays organization Admin access to dashboards.

Before:
Screenshot 2024-05-29 at 18 20 10

After:
Screenshot 2024-05-29 at 18 21 38

For context - we were not taking folder wildcard permissions into account when retrieving the list of everyone that has access to a dashboard. This change only impacts the permissions that we display - organization Admins could still access all dashboards with this bug.

**Why do we need this feature?**

To correctly display who has access to a dashboard.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/709

### CODE REVIEW

File: pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go [hunk 1]
```diff
@@ -170,14 +170,17 @@ func ProvideDashboardPermissions(
...
 			dashboard, err := getDashboard(ctx, orgID, resourceID)
 			if err != nil {
 				return nil, err
 			}
 			metrics.MFolderIDsServiceCount.WithLabelValues(metrics.AccessControl).Inc()
 			// nolint:staticcheck
-			if dashboard.FolderID > 0 {
-				query := &dashboards.GetDashboardQuery{ID: dashboard.FolderID, OrgID: orgID}
+			if dashboard.FolderUID != "" {
+				query := &dashboards.GetDashboardQuery{UID: dashboard.FolderUID, OrgID: orgID}
```
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go:183]: Note: this change is unrelated to the bug - I just updated the code to use folder UIDs not IDs (as IDs are deprecated).

File: pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go [hunk 2]
```diff
@@ -188,9 +191,13 @@ func ProvideDashboardPermissions(
...
 				}
-				return append([]string{parentScope}, nestedScopes...), nil
+
+				scopes = append(scopes, parentScope)
+				scopes = append(scopes, nestedScopes...)
+				return scopes, nil
 			}
-			return []string{dashboards.ScopeFoldersProvider.GetResourceScopeUID(folder.GeneralFolderUID)}, nil
+			scopes = append(scopes, dashboards.ScopeFoldersProvider.GetResourceScopeUID(folder.GeneralFolderUID))
+			return scopes, nil
```
Comment (kalleep) [inline pkg/services/accesscontrol/ossaccesscontrol/permissions_services.go:200]:
```suggestion
			return append(scopes, dashboards.ScopeFoldersProvider.GetResourceScopeUID(folder.GeneralFolderUID)), nil
```

**Review by kalleep — APPROVED** [review]
:rocket:
